File: solution_set1/2248. Intersection of Multiple Arrays.py
##ss
class Solution:
    def intersection(self, nums: List[List[int]]) -> List[int]:
        
        if len(nums) == 1:
            return sorted(nums[0])
        
        temp = set()
        for x in nums[0]:
            temp.add(x)
            
        def make_set(arr):
            ans = set()
            
            for x in arr:
                ans.add(x)
                
            return ans
        
        for x in range(1,len(nums)):
            #print(x,nums,temp)
            temp = temp.intersection(make_set(nums[x]))
            #print(temp)
            
       # print(temp)
        if temp is None:
            return []
        
        return sorted(list(temp))
            
        

File: cpp_solutions/2248. Intersection of Multiple Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<vector<int>>& nums) {
        
        unordered_set<int> seen;
        
        for(int x:nums[0]){
            seen.insert(x);
        }
        
        unordered_set<int> temp;
        
        for(int x=1;x<nums.size();x++){
            for(int y: nums[x]){
                if(seen.find(y)!= seen.end()){
                    temp.insert(y);
                }
            }
            seen = temp;
            temp.clear();
        }
        
        
        //set to vector conversion
        vector<int> ans;
        for(int x: seen){
            ans.push_back(x);
        }
        sort(ans.begin(),ans.end());
        return ans;
        
    }
};
